### PR TITLE
Add defaultMemberPermissions to slash commands

### PR DESCRIPTION
**Pull request by tanndlin**
Brings back the minRoleToRun functionality of legacy plugins

just a secondary precaution to prevent sensitive commands being used

### CODE REVIEW

**Comment by suneettipirneni** [comment]
We can modify our slash command structure to have a defaultPermission field. This tells discord the minimum permissions a user needs to run a command, and prevents any unsafe window between a command being deployed and an admin setting the perms.

**Comment by tanndlin** [comment]
> We can modify our slash command structure to have a defaultPermission field. This tells discord the minimum permissions a user needs to run a command, and prevents any unsafe window between a command being deployed and an admin setting the perms.

oh ok perfect, wasnt sure if that was implemented on the API side since the server can set them manually as well

File: src/common/slash.ts [hunk 1]
```diff
@@ -36,6 +38,7 @@ export interface ISlashCommand {
   name: string;
   description: string;
   options?: ApplicationCommandOptionData[];
+  minPermissions?: PermissionResolvable;
```
Comment (suneettipirneni) [inline src/common/slash.ts:41]: We can probably use the same discord name for it as it's a bit more accurate.
```suggestion
  defaultMemberPermissions?: PermissionResolvable;
```

File: src/common/slash.ts [hunk 2]
```diff
@@ -13,6 +14,7 @@ export const SlashCommand = z.object({
   name: z.string(),
   description: z.string(),
   options: z.optional(z.array(z.any())),
+  minRoleToRun: z.optional(z.any()),
```
Comment (suneettipirneni) [inline src/common/slash.ts:17]: This will need to be renamed to the new field name.

**Review by suneettipirneni — CHANGES_REQUESTED** [review]

**Review by suneettipirneni — APPROVED** [review]

**Pull request merged**